src/v/cluster/cloud_storage_size_reducer.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "cluster/cluster_utils.h"
+#include "cluster/controller_service.h"
```

Comment:
seems like it might be good enough to include "cluster/fwd.h" for many of these types, now that you've separated this class out into header/src.

Reply:
Maybe the forward declarations of newly defined classes can be in cluster/fwd.h as well

Reply:
Fixed.